File: leet_code/9_palindrome.py
class Solution(object):
    def isPalindrome(self, x):
        if x < 0:
            return False
        x_str = str(x)
        x_str_return = x_str[::-1]
        x_return = int(x_str_return)
        if x_return == x:
            return True
        else:
            return False

File: leet_code/c++/9_palindrome.cpp
#include <string>
#include <sstream>

class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        std::stringstream ss;
        ss << x;
        std::string x_str = ss.str();
        
        std::string x_rev;
        for (int i = x_str.length() - 1; i >= 0; --i) {
            x_rev.push_back(x_str[i]);
        }

        return x_str == x_rev;
    }
};
